packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -140,3 +192,97 @@ export function unstable_httpSubscriptionLink<
     };
   };
 }
+
+function createRecreateOnErrorOpts(ev: Event): RecreateOnErrorOpt {
+  if ('status' in ev && typeof ev.status === 'number') {
+    return {
+      type: 'http-error',
+      status: ev.status,
+      event: ev,
+    };
+  }
+
+  return {
+    type: 'raw',
+    event: ev,
+  };
+}
+
+/**
+ * We wrap EventSource so that is can be reinitialized with new options
+ */
+class EventSourceWrapper implements Partial<EventSource> {
+  private es: EventSource;
+
+  private listeners: Partial<
+    Record<
+      keyof EventSourceEventMap,
+      Parameters<EventSource['addEventListener']>[]
+    >
+  > = {};
+  private *getAllEventListeners() {
+    for (const _type in this.listeners) {
+      const type = _type as keyof EventSourceEventMap;
+      for (const listener of this.listeners[type] ?? []) {
```

Comment:
i'm guessing it could be
```suggestion
      const type = _type as keyof typeof this.listeners;
      for (const listener of this.listeners[type]) {
```